src/Datagrid/SimplePager.php

```diff
@@ -86,8 +86,39 @@ public function countResults(): int
         return $n;
     }
 
+    public function getCurrentPageResults(): iterable
+    {
+        if ($this->results) {
+            return $this->results;
+        }
+
+        $this->results = $this->getQuery()->execute();
+        $this->thresholdCount = \count($this->results);
+        if (\count($this->results) > $this->getMaxPerPage()) {
+            $this->haveToPaginate = true;
+
+            if ($this->results instanceof ArrayCollection) {
+                $this->results = new ArrayCollection($this->results->slice(0, $this->getMaxPerPage()));
+            } else {
+                $this->results = new ArrayCollection(\array_slice($this->results, 0, $this->getMaxPerPage()));
+            }
+        } else {
+            $this->haveToPaginate = false;
+        }
+
+        return $this->results;
+    }
+
+    /**
+     * NEXT_MAJOR: Remove this method.
```

Comment:
also add `@deprecated` to the method phpdoc?

Reply:
Indeed, added